test/nextjs.test.js

```diff
@@ -941,6 +963,26 @@ describe(`Next.js App Router: ${printPath("[test/nextjs.test.js]")}`, function (
         assert.equal(authenticatedResponse.status, 200);
         const authenticatedResponseJson = await authenticatedResponse.json();
         assert(authenticatedResponseJson.userId === process.env.user);
+
+        const requestWhereHandlerThrowsSTError = authenticatedRequest;
+        const responseWhereHandlerThrowsSTError = await withSession(
+            requestWhereHandlerThrowsSTError,
+            async (err, session) => {
+                throw new SessionError({ message: "Authentication Required!", type: "UNAUTHORISED" });
+            }
+        );
+        assert.equal(responseWhereHandlerThrowsSTError.status, 401);
+
+        const requestWhereHandlerThrowsUnknownError = authenticatedRequest;
+        const unknownError = new Error("Unknown error");
+        try {
+            await withSession(requestWhereHandlerThrowsUnknownError, async (err, session) => {
+                throw unknownError;
+            });
+            assert.fail("should not come here");
+        } catch (error) {
+            assert.strictEqual(error, unknownError);
+        }
```

Comment:
the test above already has withSession.. did you mean to use withPreParsedRequestResponse instead here?

Reply:
Great catch! I have corrected it.